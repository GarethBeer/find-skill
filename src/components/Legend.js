import React from "react";
import "../styles/Legend.css";

const Legend = () => {
  return (
    <div className="legend-container">
      <div>
        <img
          className="icon"
          src="http://labs.google.com/ridefinder/images/mm_20_green.png"
          alt="green"
        />
        <span className='info' >Professional service</span>
      </div>
      <div>
        <img
          className="icon"
          src="http://labs.google.com/ridefinder/images/mm_20_red.png"
          alt="red"
        />
        <span className='info'>Non-Professional service</span>
      </div>
    </div>
  );
};

export default Legend;
