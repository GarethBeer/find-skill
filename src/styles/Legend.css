.legend-container {
    display: flex;
    flex-direction: column;
    border: solid 1px rgb(5, 107, 5);
    padding:2px;
}
h4 {
    margin-top: 0vh;
    margin-bottom: 1vh;
    margin-left: 1vh;
}
.icon {
    margin: 0px 4px;
}
.info {
    font-size: 12px;
}
@media only screen and (max-width: 732px) {
    .legend-container {
        width: 98%;
        flex-direction: row;
        justify-content: space-evenly;
    }
}