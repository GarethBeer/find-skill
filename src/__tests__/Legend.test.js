import React from "react";
import { render } from "@testing-library/react";
import Legend from "../components/Legend";

describe("Legend", () => {
  const setUp = () => {
    const { asFragment, getByRole } = render(<Legend />);

    return {
      asFragment: asFragment(),
      greenMarker: getByRole("img", { name: "green" }),
      redMarker: getByRole("img", { name: "red" }),
    };
  };

  let renderedComponent;

  beforeEach(() => {
    renderedComponent = setUp();
  });

  it("to match the Legend snapshot", () => {
    expect(renderedComponent.asFragment).toMatchSnapshot();
  });

  it("renders correctly", () => {
    expect(renderedComponent.greenMarker).toHaveAttribute("src");
    expect(renderedComponent.greenMarker).toHaveAttribute("alt");
    expect(renderedComponent.greenMarker).toHaveClass("icon");

    expect(renderedComponent.redMarker).toHaveAttribute("src");
    expect(renderedComponent.redMarker).toHaveAttribute("alt");
    expect(renderedComponent.redMarker).toHaveClass("icon");
  });
});
