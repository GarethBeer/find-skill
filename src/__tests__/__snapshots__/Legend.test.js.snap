// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend to match the Legend snapshot 1`] = `
<DocumentFragment>
  <div
    class="legend-container"
  >
    <div>
      <img
        alt="green"
        class="icon"
        src="http://labs.google.com/ridefinder/images/mm_20_green.png"
      />
      <span
        class="info"
      >
        Professional service
      </span>
    </div>
    <div>
      <img
        alt="red"
        class="icon"
        src="http://labs.google.com/ridefinder/images/mm_20_red.png"
      />
      <span
        class="info"
      >
        Non-Professional service
      </span>
    </div>
  </div>
</DocumentFragment>
`;
